### ISSUE TITLE
Add __eq__ to BaseResolution?

### ISSUE THREAD

**Issue by bandophahita** [ScreenPyHQ/screenpy#63]
Resolution objects do not currently have the ability to check for equality nor are they hashable in a meaningful way.

```python
>>> IsVisible() == IsVisible()
False
>>>
```
Something like this should suffice:
```python
class BaseResolution:
    def __hash__(self):
        return hash((self.expected, self.line, type(self.matcher)))

    def __eq__(self, other):
        if self is other:
            return True
        elif type(self) != type(other):
            return False
        else:
            return self._members() == other._members()

    def _members(self):
        return (self.expected, self.line, type(self.matcher))
```

**Comment by bandophahita** [comment]
I'm a little uncertain just how useful this would be to anyone.  I wrote the above example when I first started using screenpy in an attempt to map tests with their equivalent selenium expected_conditions.

```python
RESOLUTION_MAP = {
        IsVisible(): EC.visibility_of_element_located,
        IsClickable(): EC.element_to_be_clickable,
        IsNot(IsVisible()): EC.invisibility_of_element_located,
        IsPresent(): EC.presence_of_element_located,
        }
```
That map felt awkward and ultimately led to adding `Eventually` which resulted in no longer needing the mapping.  
Regardless, including the ability to check for equality always seemed like a good idea.  
Or is this a case of YAGNI?
